packages/big-design/src/utils/useComponentSize.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useLayoutEffect, useState, RefObject } from 'react';
+
+interface ComponentSize {
+  height: HTMLElement['offsetHeight'];
+  width: HTMLElement['offsetWidth'];
+}
+
+const getSize = <T extends any>(element: RefObject<T>['current']): ComponentSize => ({
+  height: element ? element.offsetHeight : 0,
+  width: element ? element.offsetWidth : 0,
+});
+
+export const useComponentSize = <T extends any>(ref: RefObject<T>): ComponentSize => {
+  const [size, setSize] = useState(getSize(ref.current));
+
+  const handleResize = useCallback<() => void | MutationCallback>(() => {
+    if (ref.current) {
+      setSize(getSize(ref.current));
+    }
+  }, [ref.current]);
+
+  useLayoutEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+
+    handleResize();
+
+    if (typeof MutationObserver === 'function') {
+      const observer = new MutationObserver(handleResize);
+
+      observer.observe((ref.current as unknown) as Node, { childList: true });
+
+      return () => {
```

Comment:
🍹 `return observer.disconnect`

Reply:
React didn't like this, so I'm keeping it as is.